### ISSUE TITLE
About the world model

### ISSUE THREAD

**Issue by Rivernh** [PeidongLi/SSR#20]
Thank you for an excellent and interesting job！

I attempted to train the model locally with the same configuration as in the paper. However, the loss of the world model was difficult to reduce. Moreover, when I tried to remove the world model loss and only supervise the planning loss, the model could achieve even better results than those in the paper. I'm very puzzled. So what is the role of this world model? I don't know if there is a problem with my settings somewhere. I hope to get your help.

Best wish！

**Comment by PeidongLi** [comment]
The loss of world model reduces slightly in the latest epochs of training. I prefer it played the role of a reference model to constrain the whole model.

**Comment by Rivernh** [comment]
Thank you for your reply! However, if we don't use the world model loss , that is, without any supervision of perception, we can get better result of the metrics, I think the model mostly learned a shortcut by ego status, overfit the ego status to predict the future, so are the experimental results about the world model part reliable?

**Comment by Blake-Jiang** [comment]
> ego status

Where does the model use ego status? I can not find the code. BTW, in the paper, it claims that no ego status is used. I am just curious about your finding.

**Comment by Rivernh** [comment]
> > ego status
> 
> Where does the model use ego status? I can not find the code. BTW, in the paper, it claims that no ego status is used. I am just curious about your finding.

Oops, my bad - I totally missed that the paper specifically said they didn't use ego status. Here's the weird thing: when I tried training without any perception supervision (even the temporal self-supervision they mentioned), my model actually did better on the test set than their reported numbers. Honestly, this makes zero sense to me unless there's some sneaky ego status leakage giving the model an unfair advantage. Other papers have seen similar shortcut learning issues, so that's my best guess. Otherwise... I got nothing - this result is just straight up bizarre!

**Comment by Blake-Jiang** [comment]
> this result is just straight up bizarre

Maybe this is why open-loop result is not convincing enough for AD, closed-loop evaluation is the key.

**Comment by ShaunKung** [comment]
> > ego status
> 
> Where does the model use ego status? I can not find the code. BTW, in the paper, it claims that no ego status is used. I am just curious about your finding.

@Blake-Jiang @Rivernh 
At line 266 to 270 in file [SSR_transformer.py](https://github.com/PeidongLi/SSR/blob/main/projects%2Fmmdet3d_plugin%2FSSR%2FSSR_transformer.py#L267), the can_bus tensor of 18 scalars are projected to an 1x256 embedding and added to bev_query and then used to get bev feature.
The can bus of 18 scalars is mostly taken from pose message from nuSenes can bus expansion, as documented officially by [nuScenes](https://github.com/nutonomy/nuscenes-devkit/blob/master/python-sdk%2Fnuscenes%2Fcan_bus%2FREADME.md), the pose message contains velocity, acceleration, rotation rate and other absolute ego pose info.
I understand those can be viewed as ego status.

Hi @PeidongLi,
I can understand feeding some relative ego pose info (in ego frame) makes sense to let the model be aware of ego state. However I found that some absolute pose info, e.g. ego2global translation/rotation are also coded into can_bus vector. This seems to be a really strange operation in my view.
I'd highly appreciate if you could help to explain the intent a bit behind?
Thanks a lot!
